repondeur/zam_repondeur/views/lectures.py

```diff
@@ -135,19 +130,17 @@ def post(self) -> Response:
             LectureModel.organe == self.lecture.organe,
         ).delete()
         self.request.session.flash(("success", "Lecture supprimée avec succès."))
-        return HTTPFound(location=self.request.route_url("lectures_list"))
+        return HTTPFound(location=self.request.resource_url(self.context.__parent__))
 
 
-@view_defaults(route_name="list_amendements")
+@view_defaults(context=AmendementCollection)
 class ListAmendements:
-    def __init__(self, request: Request) -> None:
+    def __init__(self, context: AmendementCollection, request: Request) -> None:
+        self.context = context
         self.request = request
-        self.lecture = LectureModel.get(
-            chambre=request.matchdict["chambre"],
-            session=request.matchdict["session"],
-            num_texte=int(request.matchdict["num_texte"]),
-            organe=request.matchdict["organe"],
-        )
+        lecture_resource = cast(Optional[LectureResource], context.__parent__)
```

Comment:
The `cast()` is just for the static type analysis, it does not change anything at runtime.

It tells mypy about the type of an expression. A typical use would be when mypy can infer a type  that is too general, and you know the function will return a more specific subtype.

If you write `a = cast(A, expr)`, you don't say anything about the `a` variable's type, so mypy will infer it based on the type of the assigned expression, which is `A` because of the cast.

If you write `a: A = expr`, it tells mypy about the type of the variable `a`. It will then infer the type of the expression and check that it's compatible with the type of the variable. If the infered type of `expr` is not `A` or a subtype of `A`, you'll get a type error.

I'll add a comment to the code to make it clear what's happening here.

Reply:
(At least that's my understanding, but I'm still new to this so I may be mistaken about how it works exactly...)